addons/mail/models/mail_message.py

```diff
@@ -1117,34 +1098,18 @@ def message_format(self):
         message_tree = dict((m.id, m) for m in self.sudo())
         self._message_read_dict_postprocess(message_values, message_tree)
 
-        # add subtype data (is_note flag, is_discussion flag , subtype_description). Do it as sudo
-        # because portal / public may have to look for internal subtypes
-        subtype_ids = [msg['subtype_id'][0] for msg in message_values if msg['subtype_id']]
-        subtypes = self.env['mail.message.subtype'].sudo().browse(subtype_ids).read(['internal', 'description','id'])
-        subtypes_dict = dict((subtype['id'], subtype) for subtype in subtypes)
-
         com_id = self.env['ir.model.data'].xmlid_to_res_id('mail.mt_comment')
         note_id = self.env['ir.model.data'].xmlid_to_res_id('mail.mt_note')
 
-        # fetch notification status
-
-        notif_dict = defaultdict(lambda: defaultdict(list))
-        notifs = self.env['mail.notification'].sudo().search([('mail_message_id', 'in', list(mid for mid in message_tree)), ('res_partner_id', '!=', False)])
-
-        for notif in notifs:
-            mid = notif.mail_message_id.id
-            if notif.is_read:
-                notif_dict[mid]['history_partner_ids'].append(notif.res_partner_id.id)
-            else:
-                notif_dict[mid]['needaction_partner_ids'].append(notif.res_partner_id.id)
-
         for message in message_values:
+            message_sudo = message_tree[message['id']]
```

Comment:
Thanks for checking. I tried to go that far but it was more complex to handle, also due to #36263 it will increase the number of queries (o2m are currently more efficient to fetch "in the same _read" as other fields, which is only possibly by using read).

I would be ok to change that too at some point, but I just need the current PR for going forward with mail owl, I can check this again after.

Reply:
The related PR has been merged just now, so I will try again now to remove all the remaining `read`.